### ISSUE TITLE
Error while parsing a single-line block with Style/SingleLineBlockParams

### ISSUE THREAD

**Issue by jtompkins** [bbatsov/rubocop#3316]
I'm receiving an error when Rubocop parses a single-line block with no arguments.

## Expected behavior

Rubocop parses the file correctly.

## Actual behavior

An error is emitted to the console:

```
1 error occurred:
An error occurred while Style/SingleLineBlockParams cop was inspecting /Users/jtompkins/src/rbdux/spec/store_spec.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.40.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
Finished in 0.1754668439971283 seconds
```

## Steps to reproduce the problem

The line of code that's triggering the error is:

`store = Rbdux::Store.reduce(AddTodoAction) { true }`

This is happening in an RSpec test. Here's the surrounding context:

```ruby
    it 'returns the Store instance' do
      Rbdux::Action.define('add_todo')

      store = Rbdux::Store.reduce(AddTodoAction) { true }

      expect(store).to eq(Rbdux::Store.instance)
    end
```

## RuboCop version

Include the output of `rubocop -V`:

```
0.40.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```
